src/LanguageServer.Common/Utilities/DotnetInfo.cs

```diff
@@ -0,0 +1,698 @@
+                dotnetHostProcess.BeginOutputReadLine();
+                dotnetHostProcess.BeginErrorReadLine();
+
+                bool exited = dotnetHostProcess.WaitForExit(milliseconds: 5000);
+                if (!exited)
+                {
+                    logger.Error("Timed out after waiting 5 seconds for {Command} to exit.", command);
+
+                    throw new TimeoutException($"Timed out after waiting 5 seconds for '{command}' to exit.");
+                }
+
+                // Ensure redirected STDOUT and STDERROR have been flushed (tintoy/msbuild-project-tools-vscode#105).
+                logger.Debug("Waiting for redirected STDOUT/STDERR streams to complete for process {TargetProcessId}...", dotnetHostProcess.Id);
+                dotnetHostProcess.WaitForExit();
+
+                var hostExitCode = (DotNetHostExitCode)dotnetHostProcess.ExitCode;
+
+                if (hostExitCode.IsWellKnownExitCode())
+                    logger.Debug("{Command} terminated with exit code 0x{ExitCode:x} ({ExitCodeName}).", command, (int)hostExitCode, hostExitCode);
+                else
+                    logger.Debug("{Command} terminated with exit code 0x{ExitCode:x}.", command, (int)hostExitCode);
+
+                string stdOut;
+                lock (stdOutBuffer)
+                {
+                    stdOut = stdOutBuffer.ToString();
+                }
+                string stdErr;
+                lock (stdErrBuffer)
+                {
+                    stdErr = stdErrBuffer.ToString();
+                }
+
+                if (!hostExitCode.IsSuccess() || logger.IsEnabled(Serilog.Events.LogEventLevel.Verbose))
+                {
+                    if (!string.IsNullOrWhiteSpace(stdOut))
+                        logger.Debug("{Command} returned the following text on STDOUT:\n\n{DotNetOutput:l}", command, stdOut);
+                    else
+                        logger.Debug("{Command} returned no output on STDOUT.", command);
+
+                    if (!string.IsNullOrWhiteSpace(stdErr))
+                        logger.Debug("{Command} returned the following text on STDERR:\n\n{DotNetOutput:l}", command, stdErr);
+                    else
+                        logger.Debug("{Command} returned no output on STDERR.", command);
+                }
+
+                return (
+                    ExitCode: hostExitCode,
+                    StdOut: new StringReader(stdOut)
+                );
+            }
+        }
+    }
+
+    /// <summary>
+    ///     Base class for exceptions raised by <see cref="DotnetInfo"/>.
+    /// </summary>
+    public abstract class DotnetInfoException
+        : Exception
+    {
+        /// <summary>
+        ///     Create a new <see cref="DotnetInfoException"/>.
+        /// </summary>
+        /// <param name="message">
+        ///     The exception message.
+        /// </param>
+        protected DotnetInfoException(string message)
```

Comment:
If you really want all these doc comments, at least for such boilerplate members use `/// <inheritdoc />`

Reply:
Good idea! Thanks, will do.